### PR TITLE
Fix/no vpio

### PR DESCRIPTION
**Pull request by CJimenezVC**

### CODE REVIEW

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp [hunk 1]
```diff
@@ -573,6 +572,35 @@ struct iOSAudioIODevice::Pimpl      : public AudioPlayHead,
...
+        }
+
+        if (builtInMicPort) {
+            for (AVAudioSessionDataSourceDescription* source in builtInMicPort.dataSources) {
+                if ([source.orientation isEqual:AVAudioSessionOrientationBack]) {
+                    break;
+                }
+            }
+            
+            for (AVAudioSessionDataSourceDescription* source in builtInMicPort.dataSources) {
```
Comment (jamierpond) [inline modules/juce_audio_devices/native/juce_ios_Audio.cpp:622]: Should this but be in a function called `selectCardiodPolarPattern()` ?

**Review by jamierpond — APPROVED** [review]

**Comment by jamierpond** [comment]
I assume we don't care about the failing check in this case :)

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp (hunk 1 above)
Comment (CJimenezVC) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp:622]: Sure, I can separate that

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp [hunk 2]
```diff
@@ -552,9 +552,8 @@ struct iOSAudioIODevice::Pimpl      : public AudioPlayHead,
     bool setAudioPreprocessingEnabled (bool enable)
     {
         auto session = [AVAudioSession sharedInstance];
-
-        NSString* mode = (enable ? AVAudioSessionModeDefault
-                                 : AVAudioSessionModeMeasurement);
+                
+        NSString* mode = (enable ? AVAudioSessionModeVideoRecording : AVAudioSessionModeMeasurement);
```
Comment (jklein24) [inline modules/juce_audio_devices/native/juce_ios_Audio.cpp:556]: Why this change?

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp [hunk 3]
```diff
@@ -573,6 +572,52 @@ struct iOSAudioIODevice::Pimpl      : public AudioPlayHead,
...
+            }
+        }
+
+        if (builtInMicPort) {
+            for (AVAudioSessionDataSourceDescription* source in builtInMicPort.dataSources) {
+                if ([source.orientation isEqual:AVAudioSessionOrientationBack]) {
+                    break;
+                }
+            }
+        }
```
Comment (jklein24) [inline modules/juce_audio_devices/native/juce_ios_Audio.cpp]: I'm slightly confused here. Where are we actually selecting the mic choice? It kind of seems like we're just checking state, but not setting anything. Am I misunderstanding here?

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp [hunk 4]
```diff
@@ -982,6 +1027,12 @@ struct iOSAudioIODevice::Pimpl      : public AudioPlayHead,
                 
         setAnalogInputGain(0.5f);
         
+        if (!AudioIODeviceType::useDeviceVoiceProcessing) {
```
Comment (jklein24) [inline modules/juce_audio_devices/native/juce_ios_Audio.cpp]: Do we also care about isUsingBuiltInSpeaker() in the context of mic selection?

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp (hunk 2 above)
Comment (CJimenezVC) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp:556]: `AVAudioSessionModeVideoRecording` supports changing the microphone position to the top one next to the camera (unlike `AVAudioSessionModeDefault`)

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp (hunk 4 above)
Comment (CJimenezVC) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp]: True, I'll add the condition

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp (hunk 3 above)
Comment (CJimenezVC) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp]: Good catch. The mic position was changed because I'm using `AVAudioSessionModeVideoRecording`, not because of this function

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp (hunk 2 above)
Comment (CJimenezVC) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp:556]: `AVAudioSessionModeVideoRecording` has better recording quality than `AVAudioSessionModeDefault`

With `AVAudioSessionCategoryPlayAndRecord` we have 4 session modes available:
`AVAudioSessionModeDefault` --> The quality is not great, a lot of processing
`AVAudioSessionModeVoiceChat` --> The output is too low
`AVAudioSessionModeGameChat` --> Apple says to not set this mode directly
`AVAudioSessionModeVideoRecording` --> The signal processing seems to be quite minimal

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp [hunk 5]
```diff
@@ -982,6 +1041,12 @@ struct iOSAudioIODevice::Pimpl      : public AudioPlayHead,
                 
         setAnalogInputGain(0.5f);
         
+        if (isUsingBuiltInSpeaker() && !AudioIODeviceType::useDeviceVoiceProcessing) {
+            setAudioPreprocessingEnabled(true);
```
Comment (jklein24) [inline modules/juce_audio_devices/native/juce_ios_Audio.cpp:1045]: Isn't it true by default?

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp (hunk 3 above)
Comment (jklein24) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp]: So then do we still need this function?

Comment (CJimenezVC) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp]: Yes, my iPhone 6s selects the top front microphone instead of the back one so we need to force to use the back mic across all models

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp (hunk 5 above)
Comment (CJimenezVC) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp:1045]: I'm getting a different result if I don't call this function. I think `AVAudioSessionModeDefault` is the default value

File: modules/juce_audio_devices/native/juce_ios_Audio.cpp (hunk 2 above)
Comment (jklein24) [reply modules/juce_audio_devices/native/juce_ios_Audio.cpp:556]: Just a note - this will also affect the case where the flag is disabled, but I think that's ok for now anyway.

**Review by jklein24 — APPROVED** [review]
